src/ModuleList/Quiz/Quiz.tsx

```diff
@@ -1,22 +1,78 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useParams } from 'react-router-dom';
+import { SimpleGrid, Box, Flex } from '@chakra-ui/core';
+import { SubmitQuiz, CheckIcon, IncorrectIcon } from './styled';
+import { useQuizMeta } from 'src/common/api/curriculum/quiz';
+import { Question } from 'src/common/api/curriculum/quiz/requests';
 
 interface Params {
-  moduleId: string;
   quizId: string;
 }
 
+interface StringMap {
+  [key: string]: boolean;
+}
+
 const Quiz = () => {
-  const { moduleId, quizId } = useParams<Params>();
+  const { quizId } = useParams<Params>();
+
+  const { data } = useQuizMeta(Number(quizId));
+
+  const [questionIndexesAnsweredCorrectly, setQuestionIndexesAnsweredCorrectly] = useState<StringMap>({});
+  const [quizIsSubmitted, setQuizIsSubmitted] = useState<boolean>(false);
+
+  const handleAnsweredQuestion = (q: Question, qIndex: number, optionIndex: number) => {
+    return (e: React.ChangeEvent<any>) => {
+      questionIndexesAnsweredCorrectly[qIndex] = q.answer === optionIndex;
+      setQuestionIndexesAnsweredCorrectly(questionIndexesAnsweredCorrectly);
+    };
+  };
+
+  const checkQuiz = () => {
+    setQuizIsSubmitted(true);
+  };
+
+  const getLabelStyle = (q: Question, qIndex: number, optionIndex: number) => {
+    let color = '#FFFFFF';
+    if (quizIsSubmitted &&
+      !!questionIndexesAnsweredCorrectly[qIndex] === false &&
+      q.answer === optionIndex) {
+      color = '#C5A3B4';
+    }
+    return { color };
+  };
```

Comment:
Another way you could do this is to break out this logic into a styled component like.
```jsx
const QuestionLabel = styled.label`
  color: ${(props:{inIncorrect: boolean}) => props.inIncorrect ? '#C5A3B4' : '#FFFFFF'};
`;
// ...
<QuestionLabel
  htmlFor={option}
  inIncorrect={quizIsSubmitted &&
    !questionIndexesAnsweredCorrectly[qIndex] &&
    q.answer === optionIndex
  }
>
  {option}
</QuestionLabel>
```

This might be a little more readable IMO. But I'll leave the choice to you.